src/app/data-planes/views/DataPlaneDetailView.vue

```diff
@@ -9,117 +9,116 @@
     }"
     name="data-plane-detail-view"
   >
-    <AppView>
-      <template
-        v-if="warnings.length > 0"
-        #notifications
-      >
-        <ul data-testid="dataplane-warnings">
-          <!-- eslint-disable vue/no-v-html  -->
-          <li
-            v-for="warning in warnings"
-            :key="warning.kind"
-            :data-testid="`warning-${warning.kind}`"
-            v-html="t(`common.warnings.${warning.kind}`, warning.payload)"
-          />
+    <DataSource
+      v-slot="{ data: traffic, error, refresh }: TrafficSource"
+      :src="props.data.dataplaneType === 'standard' ? `/meshes/${route.params.mesh}/dataplanes/${route.params.dataPlane}/traffic` : ''"
+    >
+      <AppView>
+        <template
+          v-if="warnings.length > 0"
+          #notifications
+        >
+          <ul data-testid="dataplane-warnings">
+            <!-- eslint-disable vue/no-v-html  -->
+            <li
+              v-for="warning in warnings"
+              :key="warning.kind"
+              :data-testid="`warning-${warning.kind}`"
+              v-html="t(`common.warnings.${warning.kind}`, warning.payload)"
+            />
+            <li
+              v-if="error"
```

Comment:
Oh thanks! 🤔 but then we wouldn't show the error if the error happened to be anything but ApiError?

Considering this suggestion would adversely effect UX at the to improve of DX ever so slightly, I think I'd rather keep the `as ApiError` especially considering `Error.status` would render nothing if it was `undefined`.

---

Previous to doing this, I did have a quick go at letting you assign the error type aswell as the response type via `DataSourceResponse`, it wasn't immediately straight forwards tho, so I've left that for another time.

I just had another idea though gimme 2 tics.

Reply:
See https://github.com/kumahq/kuma-gui/pull/2015/commits/72ec94e0da5e5f4b34d0c93dad09d834e490d01e. Love that in TypeScript everything can be string 😅 , I considered just using `error` but figured `error.toString()` might be preferred.

Reply:
> Oh thanks! 🤔 but then we wouldn't show the error if the error happened to be anything but ApiError?

Which is completely correct in this case because _if_ we see a non-ApiError Error instance here, something went completely wrong and we shouldn’t be handling it in the UI. We _always_ get ApiErrors from network requests.

Meaning that

> Considering this suggestion would adversely effect UX

isn’t happening with my suggestion.

Reply:
> something went completely wrong and we shouldn’t be handling it in the UI

Why not?

If I don't see something saying "there has been an error" I'd just be sat waiting for the stats to show up? 

Considering there is no loader here either I would have no idea that I'm supposed to be seeing something here?

We would show a message in our other types of errors right? i.e. everywhere else in our UI, if `error` is set we show an `ErrorBlock`, and just briefly looking at it, while it looks like it is very very tied to ApiError it looks like it shows the `.message` for any error type? and it accepts an `Error` not an `ApiError`?

Reply:
(btw I was wondering whether we should show something different for the error anyway)

> Open to suggestions potentially clearer erroring, not totally sure on this myself.

Reply:
> If I don't see something saying "there has been an error" I'd just be sat waiting for the stats to show up?

But in all scenarios that we can reasonably expect, the user **will** get an error with the `ApiError` check! Errors that don’t fit that check are completely unexpected errors that we need to fix. They’re the null pointer exceptions and cannot read split of undefined errors that a user can’t do anything about.

We do have this generic handling as you describe but that’s really more of a convenience in the use of ErrorBlock. It’s maybe nice to show a generic "something went wrong" (and that being said, in those cases we should probably `console.error` additionally), but ultimately it’s inactionable to a user.

Using `error instanceof ApiError` perfectly fine here because it does everything you set out to handle.

Reply:
But absolutely everywhere else in the view in the entire application we catch all error types and show the UI user the error no matter if its an ApiError or just any old Error? Why is doing that not perfectly fine here?

Reply:
In the interest of getting this in, let’s resolve this.

That being said, what you wanted to do here would’ve appropriately been solved using a check for an instance of `ApiError`.

In between then and now, you changed your mind and started handling errors more generally. That wasn’t the point initially or that of the task and I fail to see what value it provides. The other mechanisms that you’re comparing to are _much_ more general in nature and so it makes sense for them to accept more broad input.

Reply:
Wait a sec, where did I change my mind?

The code expects any sort of `Error` (like the rest of the application) you wanted me to change it to only show ApiErrors specifically, I've kept it accepting any sort of `Error`. I only added the `toString` as it more explicitly turns all errors (again all errors, not just ApiErrors) into strings, as I thought your issue was that I wasn't narrowing the type at runtime. I narrowed the type to string.

> what you wanted to do here would’ve appropriately been solved using a check for an instance of ApiError.

No, it wouldn't have: "but then we wouldn't show the error if the error happened to be anything but ApiError?"

> The other mechanisms that you’re comparing to are much more general in nature and so it makes sense for them to accept more broad input.

I'm comparing to our `ErrorBlock` which we use everywhere for API interactions to show Errors, they mostly render ApiErrors but also render normal Errors. Exactly the same as what I'm doing here.

If you want different error handing it should be changed throughout the application, not just in this one tiny place.

I'm gonna hit the resolve button, please let me know if there is anything outstanding with the rest of the PR.